clap_derive/src/derives/args.rs

```diff
@@ -612,6 +637,21 @@ fn gen_parsers(
             }
         },
 
+        Ty::OptionArray => quote_spanned! { ty.span()=>
+            if #arg_matches.is_present(#name) {
+                Some(
+                    std::convert::TryInto::try_into(
+                        #arg_matches.#values_of(#name)
+                            .map(|v| v.map::<::std::result::Result<#convert_type, clap::Error>, _>(#parse).collect::<::std::result::Result<Vec<_>, clap::Error>>())
+                            .transpose()?
+                            .unwrap_or_else(Vec::new)
+                    ).unwrap()
```

Comment:
Please report an error instead of doing an `unwrap`.

Before clap 3.0 was released, we did an audit where we removed  `unwrap`s in `FromArgMatches` that assumed how the arg was setup.

Reply:
`ErrorKind::Internal` added.

Reply:
Is there a reason to add an `Internal` over using something like `WrongNumberOfValues`?  All of the other derive errors like this return the public error kind that the validator would in that circumstance.

Reply:
The error only emits when clap itself is wrong. It's unreachable.